### ISSUE TITLE
End to End Message Encryption Support in GO Client

### ISSUE THREAD

**Issue by YezenRashid** [apache/pulsar-client-go#333]
**Is your feature request related to a problem? Please describe.**
Currently End to End Encryption is not support for the Pulsar GO Client Library.  As a user of Pulsar I have many services that are in GO and Java.  I would like to leverage message encryption for all my services.  Currently it can be done with my clients written in Java, but not in GO.

**Describe the solution you'd like**
I've taken a look at the research here https://github.com/apache/pulsar/wiki/PIP-4:-Pulsar-End-to-End-Encryption
Also at the changes made for the java client library https://github.com/apache/pulsar/pull/731/files

Basically it would be the same solution implemented in GO.  I know there is support in the C++ and Java libraries.

**Describe alternatives you've considered**
I was potentially looking at using the c++ solution with GO.  I'm not sure if it would be possible.  I saw support for encryption for c++ here https://github.com/apache/pulsar/blob/master/pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h#L228

Also I know there is a CGO Library.  https://pulsar.apache.org/docs/en/client-libraries-cgo/

I'm not sure if they would be compatible though.

**Comment by YezenRashid** [comment]
I took a more in depth look.  It looks like it just needs to be exposed on the Go side.  The code exists in the c libraries.

**Comment by wolfstudy** [comment]
> I took a more in depth look. It looks like it just needs to be exposed on the Go side. The code exists in the c libraries.

Yes, If you are interested in this feature, you are always welcome to send a new pull request to implement it

**Comment by frankjkelly** [comment]
Not urgent for us but we use a lot of Go as well and would be interested in this feature. Thanks!
